src/librustdoc/clean/mod.rs

```diff
@@ -502,14 +498,17 @@ fn projection_to_path_segment<'tcx>(
 
 fn clean_generic_param_def<'tcx>(
     def: &ty::GenericParamDef,
+    defaults: ParamDefaults,
```

Comment:
I admit this could be perceived as introducing a weird coupling between `clean` and `clean::auto_trait` wrt. modularity. Previously, `auto_trait` would just `clean_ty_generics` (which calls this function) and then remove the generic parameter defaults afterwards (in a for-loop). This didn't felt a bit unclean.

Adding this param allows us to get rid of the for-loop in `auto_trait`. Furthermore, it avoids executing the rustc queries `type_of` and `const_param_default` permits us to skip a `clean_middle_ty` and a `to_string` call which boosts perf in theory.